### PR TITLE
Fix uninitialized acceleration limit, and add unit tests to TricycleDrive plugin

### PR DESCRIPTION
**Pull request by josephduchesne**

### CODE REVIEW

File: flatland_plugins/test/tricycle_drive_test.cpp [hunk 1]
```diff
@@ -62,6 +99,187 @@ TEST(TricycleDrivePluginTest, load_test) {
...
+  TricycleDrive* td = dynamic_cast<TricycleDrive*>(w->plugin_manager_.model_plugins_[0].get());
+
+
+  // Give ros a moment to set up subscribers/publishers
+  for (unsigned int i = 0; i < 100; i++) {
+    w->Update(timekeeper);
+    ros::spinOnce();
+  }
+
+  // TODO: Check odom is 0 linear, 0 angular
```
Comment (avidbots-kenneth) [inline flatland_plugins/test/tricycle_drive_test.cpp:125]: Does this TODO still apply?

File: flatland_plugins/test/tricycle_drive_test.cpp [hunk 2]
```diff
@@ -62,6 +99,187 @@ TEST(TricycleDrivePluginTest, load_test) {
...
+  EXPECT_NEAR(0, odom.twist.twist.linear.x, 0.01);
+  EXPECT_NEAR(0, odom.twist.twist.angular.z, 0.01);
+  EXPECT_NEAR(12.0, odom.pose.pose.position.x, 0.01);
+
+  // directly set the Twist message for velocity input
+  geometry_msgs::Twist cmd_vel;
+  cmd_vel.angular.x = 0;  //m/s
```
Comment (avidbots-kenneth) [inline flatland_plugins/test/tricycle_drive_test.cpp:252]: Should this be linear.x = 0?

File: flatland_plugins/test/tricycle_drive_test.cpp [hunk 3]
```diff
@@ -62,6 +99,187 @@ TEST(TricycleDrivePluginTest, load_test) {
...
+  cmd_vel.angular.z = 0.5;  // rad/s, but the limit is 0.2m/s
+  td->TwistCallback(cmd_vel);
+  for (unsigned int i = 0; i < 100; i++) {
+    w->Update(timekeeper);
+    ros::spinOnce();
+  }
+  // front wheel should have rotated to 0.2 radians, but we didn't move
+  EXPECT_NEAR(0.0, odom.twist.twist.linear.x, 0.01);
+  EXPECT_NEAR(0.0, odom.twist.twist.angular.z, 0.01);
+  EXPECT_NEAR(12.0, odom.pose.pose.position.x, 0.01);  // should have driven 0.5m from the start
```
Comment (avidbots-kenneth) [inline flatland_plugins/test/tricycle_drive_test.cpp:265]: // should not have moved from the start

File: flatland_plugins/test/tricycle_drive_test.cpp [hunk 4]
```diff
@@ -62,6 +99,187 @@ TEST(TricycleDrivePluginTest, load_test) {
...
+  EXPECT_NEAR(0, odom.twist.twist.angular.z, 0.01);
+  EXPECT_NEAR(12.0, odom.pose.pose.position.x, 0.01);
+
+  // directly set the Twist message for velocity input
+  geometry_msgs::Twist cmd_vel;
+  cmd_vel.angular.x = 0;  //m/s
+
+
+  // rotate front wheel for 1 seconds
+  cmd_vel.angular.z = 0.5;  // rad/s, but the limit is 0.2m/s
```
[the remark this answers is not in the document]
Comment (avidbots-kenneth) [reply flatland_plugins/test/tricycle_drive_test.cpp:254]: Just this one comment (the units at the end should be rad/s) otherwise PR looks good

**Review by avidbots-kenneth — APPROVED** [review]

**Review by avidbots-cameron — APPROVED** [review]

**Pull request merged**